editor/src/components/navigator/navigator-item/navigator-item-wrapper.tsx

```diff
@@ -72,72 +84,145 @@ const labelSelector = createCachedSelector(
 
 const elementWarningsSelector = createCachedSelector(
   (store: DerivedSubstate) => store.derived.elementWarnings,
-  (_: DerivedSubstate, elementPath: ElementPath) => elementPath,
-  (elementWarnings, elementPath) => {
-    return (
-      getValueFromComplexMap(EP.toString, elementWarnings, elementPath) ?? defaultElementWarnings
-    )
+  (_: DerivedSubstate, navigatorEntry: NavigatorEntry) => navigatorEntry,
+  (elementWarnings, navigatorEntry) => {
+    if (isRegularNavigatorEntry(navigatorEntry)) {
+      return (
+        getValueFromComplexMap(EP.toString, elementWarnings, navigatorEntry.elementPath) ??
+        defaultElementWarnings
+      )
+    } else {
+      return defaultElementWarnings
+    }
   },
-)((_, targetPath) => EP.toString(targetPath))
+)((_, navigatorEntry) => navigatorEntryToKey(navigatorEntry))
 
 const noOfChildrenSelector = createCachedSelector(
   (store: DerivedSubstate) => store.derived.navigatorTargets,
-  (_: DerivedSubstate, targetPath: ElementPath) => targetPath,
-  (navigatorTargets, targetPath) => {
+  (_: DerivedSubstate, navigatorEntry: NavigatorEntry) => navigatorEntry,
+  (navigatorTargets, navigatorEntry) => {
     let result = 0
     for (const nt of navigatorTargets) {
-      if (EP.isChildOf(nt, targetPath)) {
+      if (
+        isRegularNavigatorEntry(navigatorEntry) &&
+        EP.isChildOf(nt.elementPath, navigatorEntry.elementPath)
+      ) {
         result += 1
       }
     }
     return result
   },
-)((_, targetPath) => EP.toString(targetPath))
+)((_, navigatorEntry) => navigatorEntryToKey(navigatorEntry))
+
+function getNavigatorEntryLabel(
+  navigatorEntry: NavigatorEntry,
+  labelForTheElement: string,
+): string {
+  switch (navigatorEntry.type) {
+    case 'REGULAR':
+      return labelForTheElement
+    case 'CONDITIONAL_CLAUSE':
+      switch (navigatorEntry.clause) {
+        case 'then':
+          return 'Then'
+        case 'else':
+          return 'Else'
+        default:
+          throw assertNever(navigatorEntry.clause)
+      }
+    case 'SYNTHETIC': {
+      if (childOrBlockIsChild(navigatorEntry.childOrAttribute)) {
+        switch (navigatorEntry.childOrAttribute.type) {
+          case 'JSX_ELEMENT':
+            return getJSXElementNameLastPart(navigatorEntry.childOrAttribute.name)
+          case 'JSX_ARBITRARY_BLOCK':
+            return 'Unknown'
+          case 'JSX_TEXT_BLOCK':
+            return navigatorEntry.childOrAttribute.text
+          case 'JSX_FRAGMENT':
+            return 'Fragment'
+          case 'JSX_CONDITIONAL_EXPRESSION':
+            return 'Conditional'
+          default:
+            throw assertNever(navigatorEntry.childOrAttribute)
+        }
+      } else {
+        const simpleAttributeValue = jsxSimpleAttributeToValue(navigatorEntry.childOrAttribute)
+        return foldEither(
+          () => 'Unknown',
+          (value) => {
+            if (value === null) {
+              return 'null'
```

Comment:
Should this and `'undefined'` use an upper case first letter? I feel like these are using the wrong casing, but equally 'Null' and 'Undefined' feel strange to me...

Reply:
I'm going to sidestep this for now as there's other design tweaks happening.